frontend/src/components/messages/NewRequestDialog.vue

```diff
@@ -249,6 +504,93 @@ export default {
     validateDocumentsToUpload(value) {
       this.areValidFilesUploaded = value
     },
-  },
+
+    handleCheckboxChange(val, subCategory) {
+      const index = this.newRequestModel.subCategories.findIndex(item => item.subCategoryId === subCategory.subCategoryId)
+      if (index > -1) {
+        this.newRequestModel.subCategories.splice(index, 1)
+      } else if (val) {
+        this.newRequestModel.subCategories.push(subCategory)
+      }
+    },
+
+    validateChangeTypeSelection() {
+      const shouldValidate = this.isAnAccountMaintenanceRequest && !this.isAnySubCategoryChecked
+      if (shouldValidate) {
+        this.changeTypeClass = 'change-type-required'
+      }
+      return !shouldValidate
+    },
+
+    sortRequestSubCategories(subCategories, sortOrder) {
+      subCategories.sort((a, b) => {
+        let indexA = sortOrder.indexOf(a.categoryName)
+        let indexB = sortOrder.indexOf(b.categoryName)
+        if (indexA === -1) indexA = sortOrder.length
+        if (indexB === -1) indexB = sortOrder.length
+        return indexA - indexB;
+      })
+      return subCategories
+    },
+
+    async closeAssistanceRequest(assistanceRequestId) {
+      try {
+        const payload = {
+          statusCode: ASSISTANCE_REQUEST_STATUS_CODES.CLOSED_COMPLETE,
+          stateCode: CRM_STATE_CODES.INACTIVE,
+        }
+        await this.getAssistanceRequests(this.userInfo?.contactId)
```

Comment:
why do we need to get the list of assistance requests before updating a request status?

Reply:
correct, that was removed a few commits ago.

Reply:
should we remove line 526? or it is needed.

Reply:
Fixed.